agent/structs/acl.go

```diff
@@ -642,14 +681,191 @@ func (policies ACLPolicies) Merge(cache *ACLCaches, sentinel sentinel.Evaluator)
 	return acl.MergePolicies(parsed), nil
 }
 
+type ACLRoleListStub struct {
```

Comment:
Is this any different from the ACLRole struct? 

Tokens have the stub to not include secret info when listing.

Policies have the stub to not include the actual policy contents when listing. Mostly this is to not need to transfer all that data during policy replication (which we could argue could be optimized further to have custom endpoints to just send a list of id/hash combos)

Reply:
Also for roles you may be able to optimize the replication to not make any batch read requests for roles due to the original listing containing the full data.

No need to do this now of course but maybe we want a new issue to track this.

Reply:
Ah I'll nuke the list entry variant. Updating the replication to skip the secondary lookup is also pretty straightforward.